internal/service/alarms/internal/dictionary/dictionary.go

```diff
@@ -0,0 +1,190 @@
+package dictionary
+
+import (
+	"context"
+	"fmt"
+	"log/slog"
+	"sync"
+
+	monitoringv1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/selection"
+	clusterv1 "open-cluster-management.io/api/cluster/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/openshift-kni/oran-o2ims/internal/clients"
+)
+
+const (
+	managedClusterVersionLabel = "openshiftVersion-major-minor"
+	localClusterLabel          = "local-cluster"
+
+	resourceTypeCluster = "Cluster"
+)
+
+type AlarmDictionary struct {
+	Client client.Client
+
+	RulesMap map[int][]monitoringv1.Rule
+}
+
+func New(client client.Client) *AlarmDictionary {
+	return &AlarmDictionary{
+		Client:   client,
+		RulesMap: make(map[int][]monitoringv1.Rule),
+	}
+}
+
+// Load loads the alarm dictionary
+func (r *AlarmDictionary) Load(ctx context.Context) {
+	// TODO: List Resource Type from Resource Server
+	resourceTypeList := getResourceTypes()
+
+	type result struct {
+		resourceTypeID int
+		rules          []monitoringv1.Rule
+		err            error
+	}
+
+	wg := sync.WaitGroup{}
+	resultChannel := make(chan result)
+	for _, resource := range resourceTypeList {
+		wg.Add(1)
+		go func(resource resourceType) {
+			var err error
+			var rules []monitoringv1.Rule
+
+			defer func() {
+				wg.Done()
+				resultChannel <- result{
+					resourceTypeID: resource.id,
+					rules:          rules,
+					err:            err,
+				}
+			}()
+
+			switch resource.model {
+			case resourceTypeCluster:
+				rules, err = r.processCluster(ctx, resource.version)
+			default:
+				err = fmt.Errorf("unsupported resource type: %s", resource.model)
+			}
+		}(resource)
+	}
+
+	go func() {
+		wg.Wait()
+		close(resultChannel)
+	}()
+
+	for res := range resultChannel {
+		if res.err != nil {
+			slog.Error("error fetching rules for resource type", "ResourceType ID", res.resourceTypeID, "error", res.err)
+			continue
+		}
+
+		r.RulesMap[res.resourceTypeID] = res.rules
+		slog.Info("loaded rules for resource type", "ResourceType ID", res.resourceTypeID, "rules count", len(res.rules))
+	}
+
+	// TODO: Load future alarm dictionary struct
+	// TODO: Load alarm dictionary into DB
+}
+
+// processCluster processes a cluster resource type
+func (r *AlarmDictionary) processCluster(ctx context.Context, version string) ([]monitoringv1.Rule, error) {
+	cluster, err := r.getManagedCluster(ctx, version)
+	if err != nil {
+		return nil, err
+	}
+
+	promRules, err := r.getPromRules(ctx, cluster)
+	if err != nil {
+		return nil, err
+	}
+
+	// Extract rules from PrometheusRule list
+	var rules []monitoringv1.Rule
+	for _, promRule := range promRules {
+		for _, group := range promRule.Spec.Groups {
+			rules = append(rules, group.Rules...)
```

Comment:
Might be helpful to print out the alert names?

Reply:
let's do this in a future pr for filling out the dicts